storage/imx6ull-flash/imx6ull-flash-test.c

```diff
@@ -1351,6 +1351,63 @@ void test_write_read_erase_raw(void)
 }
 
 
+/* Erase whole flash, write test pattern on block zero and check if it appeared in any other block */
+void test_memory_aliasing(void)
+{
+	void *buffer = mmap(NULL, 16 * _PAGE_SIZE, PROT_READ | PROT_WRITE, MAP_UNCACHED | MAP_PHYSMEM | MAP_ANONYMOUS, -1, 0x900000);
+	flashdrv_dma_t *dma;
+	int err;
+	size_t pagesPerBlock = (flashinfo.erasesz / flashinfo.writesz);
+
+	memset(buffer, 0, 16 * _PAGE_SIZE);
+
+	for (int i = 0; i < 0x1000; ++i) {
+		((char *)buffer)[i] = 0xb2;
+		((char *)buffer)[0x1000 + i] = 0x8a;
+	}
+
+	printf("test_memory_aliasing: creating\n");
+
+	dma = flashdrv_dmanew();
+
+	printf("test_memory_aliasing: chip erase\n");
+	for (size_t i = 0; i < flashinfo.size / flashinfo.erasesz; ++i) {
+		flashdrv_erase(dma, i * flashinfo.erasesz / flashinfo.writesz);
+	}
+
+	printf("test_memory_aliasing: writing test pattern\n");
+	err = flashdrv_write(dma, 0, buffer, buffer + 0x1000);
+	printf("%d\n", err);
+
+	printf("test_memory_aliasing: test pattern verify\n");
+	err = flashdrv_read(dma, 0, buffer + 0x2000, buffer + 0x3000);
+	printf("test_memory_aliasing: error %d\n", err);
+
+	if (memcmp(buffer, buffer + 0x2000, flashinfo.writesz) != 0) {
+		printf("test_memory_aliasing: verify failed\n");
+		return;
+	}
+
+	for (size_t i = pagesPerBlock; i < flashinfo.size / flashinfo.writesz; i += pagesPerBlock) {
+		//printf("\rtest_memory_aliasing: Checking page %zu (block %zu)...", i, i / pagesPerBlock);
+		err = flashdrv_read(dma, i, buffer + 0x2000, buffer + 0x3000);
+		if (err < 0) {
+			printf("test_memory_aliasing: error %d\n", err);
+			continue;
+		}
+
```

Comment:
should skip badblocks (they might not be fully erased and have `b2`/ `8a` byte somewhere inside)

Reply:
Done